vllm/executor/gpu_executor.py

```diff
@@ -100,6 +103,31 @@ def _init_cache(self) -> None:
         # if enforce_eager is False.
         self.driver_worker.warm_up_model()
 
+    def _init_embedding(self):
+        """Initializes the maximum number of tokens can be batched by GPU.
+
+        In embedding mode, there's no need to initialize the cache engine
+        or use CUDA graph optimizations. This method calculates the maximum
+        batch size based on the GPU's capacity to handle tokens for embedding
+        purposes.
+        """
+        # Get the maximum number of tokens that can be allocated on GPU.
+        max_batch_size = self.driver_worker.profile_max_batched_tokens_for_embedding(
+        )
+
+        # NOTE(changsu): Identifying a margin threshold of 0.8 offers sufficient
+        # leeway to effectively handle memory demands while minimizing the risk
+        # of OOM errors.
+        self.scheduler_config.max_num_batched_tokens = int(
+            max_batch_size * self.model_config.max_model_len * 0.8)
```

Comment:
We might need better justification of this 0.8 factor. IMO, if changing the scheduler_config.max_num_batched_tokens is the only purpose here, it might not be suitable to put it under gpu_executor.